core/api/src/graphql/public/types/object/consumer-account.ts

```diff
@@ -206,6 +208,35 @@ const ConsumerAccount = GT.Object<Account, GraphQLPublicContextAuth>({
         )
       },
     },
+    pendingTransactions: {
+      type: GT.NonNullList(Transaction),
+      args: {
+        walletIds: {
+          type: GT.List(WalletId),
+        },
+      },
+      resolve: async (source, args) => {
+        let { walletIds } = args
+
+        if (!walletIds) {
+          const wallets = await WalletsRepository().listByAccountId(source.id)
+          if (wallets instanceof Error) {
+            throw mapError(wallets)
+          }
+          walletIds = wallets.map((wallet) => wallet.id)
+        }
```

Comment:
this logic belongs to app layer (I know in some legacy code we use services from graphql layer but we should not introduce more debt on this)